plutus-core/plutus-ir/src/PlutusIR/Mark.hs

```diff
@@ -0,0 +1,35 @@
+module PlutusIR.Mark
+    ( markNonFreshTerm
+    , markNonFreshType
+    , markNonFreshProgram
+    ) where
+
+import qualified PlutusCore.Core  as PLC
+import qualified PlutusCore.Name  as PLC
+
+import           PlutusCore.Quote
+
+import           PlutusIR.Core
+import           PlutusIR.Subst
+
+-- | Marks all the 'Unique's in a term as used, so they will not be generated in future. Useful if you
+-- have a term which was not generated in 'Quote'.
+markNonFreshTerm
```

Comment:
While we're doing this... we *should* also run these before renaming in order to handle free variables (not that we usually have them). Compare the PLC `Rename` to the PIR one.

Reply:
Yeah, still not sure what is going on with this one.  I guess I will ignore this and just leave it as it currently is (marked as TODO)...